Assets/GlitchEscape/Scripts/Misc/FloatingText.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+using TMPro;
+
+public class FloatingText : MonoBehaviour
+{
```

Comment:
make this class a singleton, ie:
```C#
public static FloatingTextController instance { 
     get { 
         if (_instance == null) { 
             _instance = GameObject.FindObjectOfType<FloatingTextController>(); 
             if (_instance == null) { Debug.LogError("no FloatingTextController instance in this scene!"); } 
        } 
        return _instance; 
}
private FloatingTextController _instance = null;
```

Note that if something could return null you should ALWAYS check the reference + issue a useful Debug.LogError message so that we can pinpoint this immediately.

And also yes, this will re-call GameObject.FindObjectOfType() and issue the Debug.LogError warning repeatedly / for every access to this class's instance property, but if this returns null we have an error anyways (ie. there is no FloatingTextController reference in the scene), ie. that should be fixed immediately.